src/exceptions.h

```diff
@@ -43,8 +43,7 @@
 #define PRETTY_FUNCTION __func__
 #endif
 
-#define throw_std_runtime_error(msg) throw std::runtime_error(std::string(msg) \
-    + " file:" + std::string(__FILE__) + " line:" + fmt::to_string(__LINE__) + " function:" + std::string(PRETTY_FUNCTION))
+#define throw_std_runtime_error(...) throw std::runtime_error(fmt::format("{} file: {} line: {} functions: {}", fmt::format(__VA_ARGS__), __FILE__, __LINE__, PRETTY_FUNCTION))
 
```

Comment:
Well, it's `functions` instead of `function`. What about turning things around: "ERROR in file {}, line {}, function {}: {}"

Reply:
Perhaps a format similar to the debug messages, nice and compact:

`"[{}:{}] {} Error: {}", __FILE__, __LINE__, PRETTY_FUNCTION, fmt::format(__VA_ARGS__))`

Reply:
Switched to this format string.